lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -239,7 +623,26 @@ export default function getAPIImplementation(): APIInterface {
             let email = formFields.filter((f) => f.id === "email")[0].value;
             let password = formFields.filter((f) => f.id === "password")[0].value;
 
-            let response = await options.recipeImplementation.signUp({ email, password, userContext });
+            let isSignUpAllowed = await AccountLinking.getInstanceOrThrowError().isSignUpAllowed({
+                newUser: {
+                    recipeId: "emailpassword",
+                    email,
+                },
+                userContext,
+            });
+
+            if (!isSignUpAllowed) {
+                return {
+                    status: "SIGNUP_NOT_ALLOWED",
+                    reason: "the sign-up info is already associated with another account where it is not verified",
+                };
+            }
+            let response = await options.recipeImplementation.signUp({
```

Comment:
call should do account linking before this

Reply:
Not really required cause `doPostSignUpAccountLinkingOperations` calls the function